narwhals/_arrow/utils.py

```diff
@@ -237,15 +237,11 @@ def extract_native(
     If one of the two sides has a `_broadcast` flag, then extract the scalar
     underneath it so that PyArrow can do its own broadcasting.
     """
-    from narwhals._arrow.dataframe import ArrowDataFrame
     from narwhals._arrow.series import ArrowSeries
 
     if rhs is None:  # pragma: no cover
         return lhs.native, lit(None, type=lhs._type)
 
-    if isinstance(rhs, ArrowDataFrame):
-        return NotImplemented
```

Comment:
this doesn't seem to affect any test

it would only be used if we supported binary operations between series and dataframes (e.g. `s - df`), which we don't anyway (and I don't think we should)